### ISSUE TITLE
Split up some of the larger scripts

### ISSUE THREAD

**Issue by vadi2** [svof/svof#446]
Some of the scripts are pretty big - should split them up so it is easier to work with them, especially for people with slow PC's.

**Comment by vadi2** [comment]
Look at how existing scripts are loaded using the loader and do the same for new ones. Note that the order of loading stuff should be kept the same, because scripts later on will make use of functions created earlier.

**Comment by PhoenixCodes** [comment]
Splitting scripts up into multiple scripts for ease of reading and editing sounds good, but at the same time consolidate modules where possible to avoid having too many base folders within the various windows? That's how I'd like to see it anyhow, what're your thoughts?

**Comment by TheLastDarkthorne** [comment]
Yeah make sense to consolidate modules instead of splitting everything, but need to keep in mind that too large scripts can make editing tricky for some people specially if they are in slow computers.
I think the way that it is now is fine though? What do you think that can be consolidated?

**Comment by vadi2** [comment]
I wouldn't consolidate anything, that goes counter to what this proposal is about?

**Comment by PhoenixCodes** [comment]
I think I may be explaining myself poorly. Most editing is (and should be) done within the Mudlet editor itself, and individual scripts are already broken up within a given module in the editor window. Having a module (like svo (aliases and triggers)) that has just the module version script in it's 'scripts', and then a separate module -just- to have 'alias functions', seems unnecessary. We can move the scripts from the one module to the other, and still have them be within their own script subheading within the module. The aliases functions not being in the alias and triggers module is what really started me thinking here, but it applies in other places too.

Same with other scripts - can break a long script up into multiple script subheadings within a single module, without creating more modules. I would think that all of the core modules could be combined under an 'svo core' module, with different folder subheadings to explain the different parts of the module. Is loading a single large module in Mudlet slower than loading multiple smaller modules?

Is there a good reason to break a module up into many smaller modules, instead of many subheadings within a single module, each subheading having it's own multiple script entries within it? The only thing I can think for this is to help people who are editing in an external editor that doesn't list scripts the way Mudlet does, but as we've seen before, people cause major bugs when they try to do too much editing outside of Mudlet... And if their computer is that slow that their editor can't handle a larger xml, perhaps Mudlet is really the best place for them to be editing scripts and such anyhow, as it does all the syntax parsing and code checking that a slower computer might struggle with?

Tl;dr: read it and consider it, I know it's long sorry.

**Comment by vadi2** [comment]
Yeah that makes sense to do. I'd put in a separate issue for it because it's a close but a it's different topic, and it confuses this ticket.

**Comment by PhoenixCodes** [comment]
Sure, just wanted to make sure that on THIS issue, what you meant was to break up actual individual scripts, not modules with tons of stuff, and that consolidating modules doesnt go against that.

**Comment by vadi2** [comment]
Correct, large individual scripts, because editing a text file that's 15k lines long is where it becomes a bit of a struggle for a low-powered pc.
